util.py

```diff
@@ -250,8 +251,16 @@ def fetchFile(source, dest):
                source[:6] == 'https:' or \
                source[:5] == 'file:' or \
                source[:4] == 'ftp:':
-            # This something that can be fetched using urllib2:
-            fd = urllib2.urlopen(source)
+            # This something that can be fetched using urllib2
+            request = source
+
+            if source.startswith('http') and '@' in source: # Auth HTTP(s)
+                url = URL(source)
+                request = urllib2.Request(url.getPlainURL())
+                b64auth = base64.standard_b64encode("%s:%s" % (url.username, url.password))
+                request.add_header("Authorization", "Basic %s" % b64auth)
```

Comment:
urllib2 has support for HTTP basic auth. Its IMO a silly interface but probably still better than hand-rolling it. See URLAccessor in repository.py for an example.